### PR TITLE
Add CACHE_DIR environment variable for managing cache

### PR DESCRIPTION
**Pull request by mederka**
This PR allows dynamic setting of a CACHE_DIR by users.

If `AUDIOLDM_CACHE_DIR` environment variable is set, it will be used instead of the default `~/.cache`.

### CODE REVIEW

File: audioldm/__init__.py [hunk 1]
```diff
@@ -6,6 +6,12 @@
 import urllib.request
 import progressbar
 
+
+CACHE_DIR = os.getenv(
+    "AUDIOLDM_CACHE_DIR",
+    os.path.join(os.path.expanduser("~"), ".cache"))
```
Comment (haoheliu) [inline audioldm/__init__.py:12]: To maintain consistency, may be this line is better to be
```python3
os.path.join(os.path.expanduser("~"), ".cache/audioldm"))
```

File: audioldm/__main__.py [hunk 2]
```diff
@@ -5,6 +5,11 @@
 
 parser = argparse.ArgumentParser()
 
+CACHE_DIR = os.getenv(
+    "AUDIOLDM_CACHE_DIR",
+    os.path.join(os.path.expanduser("~"), ".cache"))
```
Comment (haoheliu) [inline audioldm/__main__.py:10]:
```python3
os.path.join(os.path.expanduser("~"), ".cache/audioldm"))
```

File: audioldm/clap/open_clip/factory.py [hunk 3]
```diff
@@ -15,6 +15,8 @@
 
 _MODEL_CONFIG_PATHS = [Path(__file__).parent / f"model_configs/"]
 _MODEL_CONFIGS = {}  # directory (model_name: config) of model architecture configs
+CACHE_DIR = os.getenv("AUDIOLDM_CACHE_DIR", "~/.cache")
```
Comment (haoheliu) [inline audioldm/clap/open_clip/factory.py:18]:
```python3
os.path.join(os.path.expanduser("~"), ".cache/audioldm"))
```

File: audioldm/pipeline.py [hunk 4]
```diff
@@ -15,6 +15,11 @@
 import time
 
 
+CACHE_DIR = os.getenv(
+    "AUDIOLDM_CACHE_DIR",
+    os.path.join(os.path.expanduser("~"), ".cache"))
```
Comment (haoheliu) [inline audioldm/pipeline.py:20]:
```python3
os.path.join(os.path.expanduser("~"), ".cache/audioldm"))
```

**Comment by haoheliu** [comment]
@mederka Nice feature! Thanks for the PR. A few comments were posted, otherwise looks good!

**Comment by mederka** [comment]
@haoheliu, thank you for your comments. The suggested changes have been pushed.

**Pull request merged**